### PR TITLE
Shared disk transmitter use internal mime

### LINKED ISSUE #1964: SharedDiskRecordTransmitter assumes JSON for numerical records
**Describe the bug**
The e.g. `dump` method of the transmitter allows the MIME to vary, but the file on disk is JSON and `dump`'s implementation is basically a copy operation. So even though you call `dump(r, "text/csv")` you'll get JSON.

Storing the file on disk is supposed to be an implementation detail, so JSON on disk isn't the problem in itself.

### PR DESCRIPTION
**Pull request by sondreso**
**Issue**
Resolves #1964 

**Approach**
Add internal mime type for `SharedDiskTransmitter`. Add YAML serializer to actually get usage of two different formats.

### CODE REVIEW

File: ert/data/_record.py [hunk 1]
```diff
@@ -351,10 +354,11 @@ async def _transmit_blob_record(self, record: BlobRecord) -> str:
     async def _load_numerical_record(self) -> NumericalRecord:
         async with aiofiles.open(str(self._uri), mode="rt", encoding="utf-8") as f:
             contents = await f.read()
+        serializer = get_serializer(SharedDiskRecordTransmitter._INTERNAL_MIME_TYPE)
         if self._record_type == RecordType.MAPPING_INT_FLOAT:
-            data = json.loads(contents, object_hook=parse_json_key_as_int)
+            data = serializer.decode(contents)
         else:
-            data = json.loads(contents)
+            data = serializer.decode(contents)
```
Comment (jondequinor) [inline ert/data/_record.py:361]: Unnecessary branching? Both cases are equal. They didn't used to be, but you mentioned you made an issue for that. Could you add that to a `# TODO`?

Comment (jondequinor) [reply ert/data/_record.py:361]: Is YAML respecting the mapping from int to float?

Comment (sondreso) [reply ert/data/_record.py:361]: Yes, forgot to clean this up now that it was switched to yaml! And yes yaml does not have the same problem as JSON, as there is a difference between the key `"10"` and `10`:

```
>>> yaml.load(yaml.dump({10:10}))
{10: 10}
>>> yaml.load(yaml.dump({"10":10}))
{'10': 10}
```

**Review by jondequinor — APPROVED** [review]
Great!

**Pull request merged**